site/docs/latest/admin/Upgrade.md

```diff
@@ -0,0 +1,152 @@
+3. Upgrade Proxies
+  1. Canary: canary an upgraded version in one or small set of proxies.
+  2. Rolling Upgrade: rollout the upgraded version to all proxies in the cluster once you determined a version is safe after canary.
+
+Typically you don't need to upgrade zookeeper unless there are strong reasons documented in "Upgrade Guides" section below.
+But if you would like to keep your zookeeper cluster consistent with other components, follows the general upgrade guideline as others.
+
+1. Canary: canary an upgraded version in one of the zookeeper servers.
+2. Rolling Upgrade: rollout the upgraded version to all zookeeper servers incrementally, one at a time.
+
+## Upgrade Bookies
+
+> You can also read Apache BookKeeper [Upgrade Guide](http://bookkeeper.apache.org/docs/latest/admin/upgrade) for more details.
+
+### Canary
+
+It is wise to canary an upgraded version in one or small set of bookies before upgrading all bookies in your live cluster.
+
+You can follow below steps on how to canary a upgraded version:
+
+1. Stop a Bookie.
+2. Upgrade the binary and configuration.
+3. Start the Bookie in `ReadOnly` mode. This can be used to verify if the Bookie of this new version can run well for read workload.
+   ```shell
+   bin/pulsar bookie --readOnly
+   ```
+4. Once the Bookie is running at `ReadOnly` mode successfully for a while, restart the Bookie in `Write/Read` mode.
+   ```shell
+   bin/pulsar bookie
+   ```
+5. After step 4, the Bookie will serve both write and read traffic.
+
+#### Rollback Canaries
+
+If problems occur during canarying an upgraded version, you can simply take down the problematic Bookie node. The remain bookies in the old cluster
+will repair this problematic bookie node by autorecovery. Nothing needs to be worried about.
+
+### Upgrade Steps
+
+Once you determined a version is safe to upgrade in a few bookies in your cluster, you can perform following steps to upgrade all bookies in your cluster.
+
+1. Decide on performing a rolling upgrade or a downtime upgrade.
+2. Upgrade all Bookies (more below)
+
+#### Upgrade Bookies
+
+In a rolling upgrade scenario, upgrade one Bookie at a time. In a downtime upgrade scenario, take the entire cluster down, upgrade each Bookie, then start the cluster.
+
+For each Bookie:
+
+1. Stop the bookie. 
+2. Upgrade the software (either new binary or new configuration)
+2. Start the bookie.
+
+## Upgrade Brokers / Proxies
+
+Brokers and Proxies are almost same. They are `stateless` comparing to Bookies. so it is very straightforward on upgrading these two services.
+
+### Canary
+
+Although they are `stateless` and easier to upgrade, it is still wise to canary an upgraded version in one or small set of brokers/proxies before upgrading
+all brokers in your live cluster.
+
+You can follow below steps on how to canary a upgraded version:
+
+1. Stop a Broker (or Proxy).
```

Comment:
There seems to be a lot of duplication here in upgrading canary and upgrading all brokers. The procedure is the same in both cases. We should try to cut this document down to as few words as possible, because otherwise people's eyes will glaze over when reading and they'll miss something important.